src/NLog/Internal/FilePathLayout.cs

```diff
@@ -164,6 +164,13 @@ private string GetRenderedFileName(LogEventInfo logEvent, System.Text.StringBuil
 
             if (reusableBuilder != null)
             {
+                if (!_layout.IsThreadAgnostic)
```

Comment:
IsThreadAgnostic means that it doesn't have to be precalculated by async-wrappers or buffering-wrappers. As none of the layouts requires the context-information of the logging-thread.

But it it requires the context-information of the logging-thread, and there is a async-wrapper or buffering-wrapper. Then it is possible to extract the precalculated layout-rendering-result.